crypto/rsa_extra/rsa_test.cc

```diff
@@ -653,7 +654,9 @@ TEST(RSATest, GenerateSmallKey) {
   EXPECT_EQ(ERR_LIB_RSA, ERR_GET_LIB(err));
   EXPECT_EQ(RSA_R_KEY_SIZE_TOO_SMALL, ERR_GET_REASON(err));
 }
-#else
+
+#elif !defined(OPENSSL_IOS) // ASSERT_DEATH is not defined for iOS.
```

Comment:
We might want to consider using `ASSERT_DEATH_IF_SUPPORTED` to simplify this, but I can see not wanting to use that since it will just lead to some wasted cycles running tests on IOS. I only suggest it since it seems to be a way of dealing with these compatibility issues in the future (though I'm not sure there are any real issues like this we will hit). Certainly not a blocker just food for thought.

https://github.com/google/googletest/blob/main/googletest/include/gtest/gtest-death-test.h#L327

Reply:
good find! I changed the asserts to `ASSERT_DEATH_IF_SUPPORTED`